driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientElement.java

```diff
@@ -55,6 +56,8 @@ public Graph graph() {
     }
 
     public <V> Property<V> property(final String key, final V value) {
+        if(value == null)
+            throw Property.Exceptions.propertyKeyCanNotBeNull();
```

Comment:
OrientDB can handle nulls, tinkerpop can not. 

Who has the final say?

May be we need a patch on gremlin to made this a feature